slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/SlashingProtection.java

```diff
@@ -38,5 +39,7 @@ boolean maySignBlock(
 
   void importData(InputStream output);
 
+  void importDataWithFilter(InputStream output, Optional<List<String>> pubkeys);
```

Comment:
I made it an optional to differentiate between passing an empty list (filters out everything) and not passing anything (doesn't filter anything). I use it the `InterchangeV5Importer` implementation to re-use the same logic. 

I guess in the external API since it's a separate method the Optional feels superfluous though, I'll remove it.

Reply:
ahah, got it now, thanks for explaining. Optional.of(emptyList) is not an obvious use case IMO. Perhaps a nested KeyFilter class with a static factory method like KeyFilter.noFilter() might make the intended behaviour clearer to the reader?

At the very least, I think you can limit the scope even further to hide this complexity, i.e. in InterchangeV5Importer:
```java
public void importData(final InputStream input) throws IOException {
    importData(input, Optional.empty());
}

public void importDataWithFilter(final InputStream input, final List<String> pubkeys)
    throws IOException {
  importData(input, Optional.of(pubkeys));
}

public void importData(final InputStream input, final Optional<List<String>> pubkeys)
    throws IOException {
...
```

Reply:
that's exactly what i did in the latest commit :)